src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -239,12 +239,7 @@ protected virtual void ProduceTagsSynchronously(TaggerContext<TTag> context, Doc
             //
             // Taggers that need to be called accurately should override this method to produce
             // results quickly if possible.
-            //
-            // We use JTF.Run here as our tag sources use JTF to quickly switch to the UI thread to report their initial
-            // set of tags.  We need JTF.Run here to ensure we don't deadlock if we're waiting on them while they're
-            // doing that notification.  Specifically, the TagSource will report its first set of tags immediately to the 
-            // UI thread using a JTF.Run and a SwitchToMainThread call.  If we do not use JTF here, we will deadlock.
-            this.ThreadingContext.JoinableTaskFactory.Run(() => ProduceTagsAsync(context, spanToTag, caretPosition));
+            ProduceTagsAsync(context, spanToTag, caretPosition).Wait(context.CancellationToken);
```

Comment:
I thought this was necessary, as the classification tagger the first time may still call into the VS service to get the workspace status task, and that could implicitly require the UI thread.

Reply:
talked offline.  The code in the lastest commit is safe to this.  all the work to access the VS service is pushed off to a different thread, and we only ever examient he result of this by checking the .IsComplete bit on the task.  We do *not* await/await/.result that task.  So even if our caller does a .Wait, there is no risk of deadlock as we're only ever making a free-threaded, non-blocking check from the code that is waited.